ambry-router/src/test/java/com/github/ambry/router/MockServer.java

```diff
@@ -445,6 +471,25 @@ private UndeleteResponse makeUndeleteResponse(UndeleteRequest undeleteRequest, S
     }
   }
 
+  /**
+   * Make a {@link ReplicateBlobResponse} for the given {@link ReplicateBlobRequest} for which the given {@link ServerErrorCode} was
+   * encountered.
+   * *
+   * @return the constructed {@link ReplicateBlobResponse}
+   */
+  ReplicateBlobResponse makeReplicateBlobResponse(ReplicateBlobRequest replicateBlobRequest,
+      ServerErrorCode replicateBlobError) {
+    if (replicateBlobError == ServerErrorCode.No_Error) {
+      replicateBlobError =
+          errorCodeForBlobs.getOrDefault(replicateBlobRequest.getBlobId().getID(), ServerErrorCode.No_Error);
+    }
+    if (replicateBlobError == ServerErrorCode.No_Error) {
+      updateBlobMap(replicateBlobRequest);
```

Comment:
Can we combine them together?

Reply:
Inside the first if block, we'll re-assign the replicateBlobError again. Since replicateBlobError is changed, if I combine these two, I still need add the inner if condition to test the replicateBlobError again.
Seems it won't have much difference. I hope it's ok that I don't change it.



if (replicateBlobError == ServerErrorCode.No_Error) {
      replicateBlobError =
          errorCodeForBlobs.getOrDefault(replicateBlobRequest.getBlobId().getID(), ServerErrorCode.No_Error);
}
if (replicateBlobError == ServerErrorCode.No_Error) {
      updateBlobMap(replicateBlobRequest);
}